logs_sdk/lib/opentelemetry/sdk/logs/log_record.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Logs
+      # Implementation of OpenTelemetry::Logs::LogRecord that records log events.
+      class LogRecord < OpenTelemetry::Logs::LogRecord
+        attr_accessor :timestamp,
+                      :observed_timestamp,
+                      :span_context,
+                      :severity_text,
+                      :severity_number,
+                      :body,
+                      :resource,
+                      :instrumentation_scope,
+                      :attributes
+
+        # Creates a new {LogRecord}.
+        #
+        # @param [optional Float, Time] timestamp Time when the event occurred.
+        # @param [optional Float, Time] observed_timestamp Time when the event
+        #   was observed by the collection system. If nil, will first attempt
+        #   to set to `timestamp`. If `timestamp` is nil, will set to
+        #   `Process.clock_gettime(Process::CLOCK_REALTIME, :nanosecond)`.
+        # @param [optional OpenTelemetry::Trace::SpanContext] span_context The
+        #   OpenTelemetry::Trace::SpanContext to associate with the
+        #   {LogRecord}.
+        # @param [optional String] severity_text The log severity, also known as
+        #   log level.
+        # @param [optional Integer] severity_number The numerical value of the
+        #   log severity.
```

Comment:
The API documentation for [Logger#emit expects the `Context` to be passed.](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#emit-a-logrecord)

However, the [LogRecord data model](https://opentelemetry.io/docs/specs/otel/logs/data-model/#log-and-event-record-definition) doesn't include the context. It does include the `span_id`, `trace_id` and `trace_flags`, which are all available in the SpanContext.

I'm not exactly sure what's the right call here. Should I add `Context` as well? Would the `span_id`, `trace_id`, and `trace_flags` exist in the context? Would the other `Context` key/value pairs go into attributes?